Analysis/analysis.r

```diff
@@ -113,27 +174,47 @@ obese_df %>%
 
 # confidence interval
 
-dclus2 <- svydesign(id=~id, weights = ~wt_int, data=obese_df)
+dclus2 <- svydesign(id=~id, weights = ~wt_int, data=full)
 
-funDiff <- function(bmi_class_c, sex_c){
+funDiff_byBMI <- function(bmi_class_c, sex_c){
     sub <- subset(dclus2, sex == sex_c & bmi_class == bmi_class_c)
     tt<-svyttest(I(intake_end - intake_start)~0, sub)
     return(data.frame(sex = sex_c, bmi_class = bmi_class_c, confint(tt, level=0.95)))
-    }
+}
+
+funDiff_excess <- function(sex_c){
+  sub <- subset(dclus2, sex == sex_c & bmi_class %in% c("overweight", "obese", "morbidly obese"))
+  tt<-svyttest(I(intake_end - intake_start)~0, sub)
+  return(data.frame(sex = sex_c, confint(tt, level=0.95)))
+}
+
+funDiff_all <- function(sex_c){
```

Comment:
A general comment on the functions, I'm not sure if its a thing in R (please ignore if its not) but I think it would be helpful to do the following to make the script a little easier to follow:

1. Have all functions at the top or even in a separate script thats read in
2. Give functions docstrings that explain what they do / what to expect as inputs and outputs